pyproject.toml

```
@@ -2,45 +2,230 @@
 # Distributed under the terms of the Modified BSD License.
 
 [build-system]
-requires=["jupyter_packaging>=0.9,<2"]
-build-backend = "jupyter_packaging.build_api"
+requires = ["hatchling>=1.2.0"]
+build-backend = "hatchling.build"
 
-[license]
-file="LICENSE"
+[project]
+name = "jupyterlab"
+description = "JupyterLab computational environment"
+readme = "README.md"
+license = ""
+requires-python = ">=3.7"
+authors = [
+    { name = "Jupyter Development Team", email = "jupyter@googlegroups.com" },
+]
+keywords = [
+    "ipython",
+    "jupyter",
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Framework :: Jupyter",
+    "Framework :: Jupyter :: JupyterLab",
+    "Framework :: Jupyter :: JupyterLab :: 4",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+]
+dependencies = [
+    "importlib-metadata>=3.6;python_version<\"3.10\"",
+    "importlib-resources>=1.4;python_version<\"3.9\"",
+    "ipykernel",
+    "jinja2>=3.0.3",
+    "jupyter_core",
```

@fcollonval  the lsp server extension is missing.

```suggestion
    "jupyter_core",
    "jupyter-lsp>=1.5.1",
```